src/ZipkinBundle/Components/Messenger/ZipkinReceiveHandler.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+
+namespace ZipkinBundle\Components\Messenger;
+
+use Symfony\Component\Messenger\Envelope;
+use Zipkin\Propagation\B3;
+use Zipkin\Propagation\Getter;
+use Zipkin\Kind;
+use Zipkin\Tracing;
+
+class ZipkinReceiveHandler implements ZipkinHandlerInterface
+{
+    /**
+     * @var \Zipkin\Tracer
+     */
+    private $tracer;
+    /**
+     * @var B3
+     */
+    private $b3;
+
+    public function __construct(Tracing $tracing, B3 $b3)
```

Comment:
You don't need to pass $b3 in here as $tracing->getPropagation() is going to give it to you :).

Reply:
Done